network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/state/NetworkProtectionStateImpl.kt

```diff
@@ -29,7 +30,12 @@ import kotlinx.coroutines.launch
 class NetworkProtectionStateImpl @Inject constructor(
     private val vpnFeaturesRegistry: VpnFeaturesRegistry,
     private val coroutineScope: CoroutineScope,
+    private val netPWaitlistRepository: NetPWaitlistRepository,
 ) : NetworkProtectionState {
+    override fun isAuthenticated(): Boolean {
+        return netPWaitlistRepository.getAuthenticationToken() != null
+    }
+
```

Comment:
Ditto in previous comment. I'd rename this to `isOnboarded()` and the implementation would be

```kotlin
    override suspend fun isOnboarded(): Boolean = withContext(dispatcherProvider.io()) {
        return@withContext netpCohortStore.cohortLocalDate != null
    }
```